### PR TITLE
Add ability to create (POST) and delete resources

### PR DESCRIPTION
**Pull request by fanqiuwen**
This makes it possible to create a new domain and delete an existing one.

`create` is a class method that takes in keyword arguments to be POSTed as json.
`delete` is a regular method that can be called on an instance of a resource.

Example usage:

    >>> example = strongarm.Domain.create(name='example.com')
    >>> example
    Domain({u'date': u'2015-08-05T21:08:06.316441Z', u'name': u'example.com', u'user': u'owen'})
    >>> example.delete()

Todo:
- [x] Discuss architecture
- [x] Update documentation
- [x] Improve test coverage

### CODE REVIEW

File: strongarm/common.py [hunk 1]
```diff
@@ -179,3 +211,34 @@ class ListableResource(object):
...
+    The `delete` method returns None on successful deletion.
+
+    """
+
+    # The attribute to be used as the unique identifier.
+    id_attr = 'id'
+
+    def delete(self, **kwargs):
+        endpoint = strongarm.host + self.endpoint + str(getattr(self, self.id_attr))
+        request('delete', endpoint)
```
Comment (fanqiuwen) [inline strongarm/common.py]: `delete` simply makes a request to the API (with error checking). It doesn't do anything special on success. In particular, the instance of the resource is still intact in Python, and anyone holding onto a previously generated list of resources still thinks this instance exists. Suggestions are welcome on a more robust way to handle this.

Comment (clokep) [inline strongarm/common.py]: I don't think there's anything you can do about that. We don't have any global resource list or anything to mark dirty. This architecture seems OK to me.

Comment (dicato) [inline strongarm/common.py]: What happens if you try to delete a domain that does not exist?

Comment (clokep) [inline strongarm/common.py]: My expectation is that it would `404`?

Comment (fanqiuwen) [inline strongarm/common.py]: Yes, it'll `404`.

Comment (dicato) [inline strongarm/common.py]: Great, so the caller at least gets some indication of success via the HTTP response code. Makes sense to me.

**Pull request merged**